tests/integration/optimizers/test_optimizer.py

```diff
@@ -16,56 +19,80 @@
+    os.environ['JINA_OPTIMIZER_FLOW_JAML_PATH'] = os.path.join(root, 'flow.yml')
+    os.environ['JINA_OPTIMIZER_DATA_PATH'] = os.path.join(root, 'data.jsonlines')
+    os.environ['JINA_OPTIMIZER_PARAMETER_PATH'] = os.path.join(root, 'parameter.yml')
+    os.environ['JINA_OPTIMIZER_WORKSPACE_DIR'] = str(tmpdir)
+    os.environ['JINA_OPTIMIZER_OUTPUT_FILE'] = os.path.join(tmpdir, 'best_parameters.yml')
+    yield
+    del os.environ['JINA_OPTIMIZER_FLOW_JAML_PATH']
+    del os.environ['JINA_OPTIMIZER_DATA_PATH']
+    del os.environ['JINA_OPTIMIZER_PARAMETER_PATH']
+    del os.environ['JINA_OPTIMIZER_WORKSPACE_DIR']
+    del os.environ['JINA_OPTIMIZER_OUTPUT_FILE']
+
+
 def document_generator(num_doc):
     for _ in range(num_doc):
         doc = Document(content='hello')
         groundtruth_doc = Document(content='hello')
         yield doc, groundtruth_doc
 
 
-def test_optimizer(tmpdir):
+@pytest.mark.parametrize('use_output_file', (True, False))
+def test_optimizer(tmpdir, use_output_file):
     eval_flow_runner = SingleFlowRunner(
-        flow_yaml='tests/integration/optimizers/flow.yml',
+        flow_yaml=os.path.join('tests', 'integration', 'optimizers', 'flow.yml'),
         documents=document_generator(10),
         request_size=1,
         task='search',
     )
-
+    output_file = os.path.join(tmpdir, 'results', 'best_parameters.yml')
+    output_file_param = output_file if use_output_file else None
     opt = FlowOptimizer(
         flow_runner=eval_flow_runner,
-        parameter_yaml='tests/integration/optimizers/parameter.yml',
+        parameter_yaml=os.path.join('tests', 'integration', 'optimizers', 'parameter.yml'),
         evaluation_callback=MeanEvaluationCallback(),
         workspace_base_dir=str(tmpdir),
+        output_file=output_file_param,
         n_trials=5,
     )
-
     result = opt.optimize_flow()
-    result_path = str(tmpdir) + '/results/best_parameters.yml'
-    result.save_parameters(result_path)
-    parameters = result.best_parameters
+    assert result.best_parameters == BEST_PARAMETERS
+    validate_results(output_file, use_output_file)
+
 
-    assert parameters == BEST_PARAMETERS
-    assert yaml.load(open(result_path)) == BEST_PARAMETERS
+def validate_results(output_file, use_output_file):
+    if use_output_file:
+        assert yaml.load(open(output_file)) == BEST_PARAMETERS
+    else:
+        with pytest.raises(FileNotFoundError):
+            _ = open(output_file)
 
 
-def test_yaml(tmpdir):
+@pytest.mark.parametrize('use_output_file', (True, False))
+def test_yaml(tmpdir, use_output_file):
     jsonlines_file = os.path.join(tmpdir, 'docs.jsonlines')
+    output_file = os.path.join(tmpdir, 'results', 'best_parameters.yml')
     optimizer_yaml = f'''!FlowOptimizer
 version: 1
 with:
   flow_runner: !SingleFlowRunner
     with:
-      flow_yaml: 'tests/integration/optimizers/flow.yml'
+      flow_yaml: {os.path.join('tests', 'integration', 'optimizers', 'flow.yml')}
```

Comment:
does it need to be absolute? this will not work from other places than `jina` root right?

Reply:
yes the test needs to be run from the root

Reply:
is there an option to have it `relative` to the `current` file? the same way other `tests` do?